File: cpg-language-typescript/src/test/resources/typescript/visibility.ts
class TsClass {
    public publicField: number;
    protected protectedField: number;
    private privateField: number;
    static staticField: number;
    private static privateStaticField: number;
    defaultField: number;
    #hardField: number;
    static #hardStaticField: number;

    private constructor() {}

    public publicMethod(): void {}
    protected protectedMethod(): void {}
    private privateMethod(): void {}
    static staticMethod(): void {}
    private static privateStaticMethod(): void {}
    #hardMethod(): void {}
}

File: cpg-language-jvm/src/test/resources/class/visibility/mypackage/Visibility.java
package mypackage;

/**
 * A fixture exercising every JVM access-flag combination the frontend maps onto the canonical
 * visibility model. The class itself is {@code public}; the nested {@link PackagePrivate} class has
 * no access flag and is therefore package-private.
 */
public class Visibility {

    public int publicField;
    protected int protectedField;
    private int privateField;
    int packageField;
    public static int staticField;

    public void publicMethod() {}

    protected void protectedMethod() {}

    private void privateMethod() {}

    void packageMethod() {}

    public static void staticMethod() {}
}

class PackagePrivate {
    int value;
}
